postgres_kernel/kernel.py

```diff
@@ -98,17 +98,60 @@ def fetchall(self, query):
             return None, None
 
     CONN_STRING_COMMENT = re.compile(r'--\s?connection:\s?(.*)$')
+    AUTOCOMMIT_SWITCH = re.compile(r'--autocommit:(.*)$')
 
     def change_connection(self, conn_string):
         self._conn_string = conn_string
         self._start_connection()
 
+    def switch_autocommit(self, switch_to):
+        if not self._conn:
+            self._start_connection()
+        self._conn.commit()
+        self._conn.autocommit = switch_to
+
+    def handle_autocoomit_switch(self, switch):
+        """
+        Strip and make a string case insensitive and ensure it is either 'true' or 'false'.
+
+        If neither, prompt user for either value.
+        When 'true', return True, and when 'false' return False.
+        """
+        parsed_switch = switch.strip().lower()
+        if not parsed_switch in ['true', 'false']:
+            self.send_response(
+                self.iopub_socket, 'stream', {
+                    'name': 'stderr',
+                    'text': 'autocommit must be true or false.\n\n'
+                    }
+                )
+            return {'status': 'error', 'execution_count': self.execution_count}
+
+        switch_bool = (parsed_switch == 'true')
+        self.switch_autocommit(switch_bool)
+        self.send_response(
+            self.iopub_socket, 'stream', {
+                'name': 'stdout',
+                'text': (
+                    f'commited current transaction & switched autocommit mode to '
+                    f'{str(self._conn.autocommit)}'
```

Comment:
I believe that, with the exception of these two f-strings, we're compatible with both Python 2 and 3. Mind if we remove these in favor of boring old `.format()`?

Reply:
meh fiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiine